src/java.desktop/share/classes/javax/swing/JSplitPane.java

```diff
@@ -369,22 +369,24 @@ public JSplitPane(int newOrientation,
      */
     @Override
     public void setComponentOrientation(ComponentOrientation orientation) {
-        super.setComponentOrientation(orientation);
-        Component leftComponent = this.getLeftComponent();
-        Component rightComponent = this.getRightComponent();
-        if (!this.getComponentOrientation().isLeftToRight()) {
-            if (rightComponent != null) {
-                setLeftComponent(rightComponent);
-            }
-            if (leftComponent != null) {
-                setRightComponent(leftComponent);
-            }
-        } else {
-            if (leftComponent != null) {
-                setLeftComponent(leftComponent);
-            }
-            if (rightComponent != null) {
-                setRightComponent(rightComponent);
+        if (!orientation.equals(this.getComponentOrientation())) {
+            super.setComponentOrientation(orientation);
```

Comment:
I would like to bring up for discussion whether we should move this line outside the if block. From what I can tell, it was always executed before the initial fix, so this change might affect the original behavior.

Reply:
If we call setComponentOrientation in test (like in TestSplitPaneOrientationTest.java which was added for JDK-4265389) then this check would fail as ComponentOrientation object would change if we call super.setComponentOrientation outside the if block and so left/right component would not be set..

Reply:
But we can read/save orientation, then call super, then make a check? 
I am not sure how important to always call super.setComponentOrientation(orientation);
can we miss some notification fired from that method?

Reply:
I guess all concerns are addressed..If there is no proven concern on this, I would integrate this.